### PR TITLE
Lab 5

### PR DESCRIPTION
**Pull request by danylykkk**

### CODE REVIEW

File: Lab5.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    def Power(self):
+        return 160
+
+def GetPower (car):
+    car.Power()
+
+
+#ISP
+
+class ICircle:
```
Comment (polyakovartem) [inline Lab5.py]: I don't see interface segregation here

File: Lab5.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        pass
+
+class Triangle(ITriangle):
+    def DrawTriangle(self):
+        #code
+        pass
+
+
+
+#DIP
```
Comment (polyakovartem) [inline Lab5.py]: this is not dependency injection

**Review by polyakovartem — CHANGES_REQUESTED** [review]

File: Lab5.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    def __init__ (self, db: Idb):
+        self.db = db
+
+    def GetBooks (self):
+        conn = self.db.Connect()
+
+mysql = MySQLDB()
+mongo = MongoDB()
+b1 = Books(mysql).GetBooks()
+b2 = Books(mongo).GetBooks()
```
Comment (polyakovartem) [inline Lab5.py]: https://docs.python.org/3/library/__main__.html

Comment (polyakovartem) [reply Lab5.py]: please also add the usage examples for every principle

**Review by polyakovartem — CHANGES_REQUESTED** [review]

File: Lab5.py [hunk 4]
```diff
@@ -8,12 +8,20 @@ def Greeting (self):
         print("Hello, my name is " + self.name)
 
     #def Save (self):
-    #    pass
+    #    print("Saved")
```
Comment (polyakovartem) [inline Lab5.py]: this is not the intention of this function to print

**Comment by polyakovartem** [comment]
please make your samples real

**Review by polyakovartem — CHANGES_REQUESTED** [review]

File: Lab5.py (hunk 4 above)
Comment (danylykkk) [reply Lab5.py]: The task is to show SOLID principles and I show it with print element to show what method is running. You want me to make real save algoritm? If yes, why? The save algorithm is not part of SOLID principles.

Comment (danylykkk) [reply Lab5.py]: Am I right?

Comment (polyakovartem) [reply Lab5.py]: look, you've done enough to get a mark, but in case you want to learn anything you have to do extra

Comment (polyakovartem) [reply Lab5.py]: and btw, in case you don't know, implementing this function should've taken the same time as answering me in this pull request

**Review by polyakovartem — CHANGES_REQUESTED** [review]
what about fixing other comments?

File: Lab5.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            port = 3306
+        )
+
+    def SavePerson (self, person: Person):
+        cur = con.cursor()
+        cur.execute("INSERT INTO persons (ID, NAME) VALUES (NULL, %s)", (person.name))
+        self.con.commit()
+        cur.close()
+
+    def __exit__ (self):
```
Comment (polyakovartem) [inline Lab5.py]: why do you need this method?

**Review by polyakovartem — CHANGES_REQUESTED** [review]
please fix all PEP8 violations

File: Lab5.py (hunk 5 above)
Comment (danylykkk) [reply Lab5.py]: __exit__ closes connection to database when object destroy
__init__ initializes connection to database
SavePerson inserts data to database

Comment (polyakovartem) [reply Lab5.py]: can you please demo the conditions under which exit is called

Comment (danylykkk) [reply Lab5.py]: image

**Review by polyakovartem — APPROVED** [review]

**Pull request closed without merging**